opnode/l2/sync_reference.go

```diff
@@ -0,0 +1,55 @@
+package l2
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/eth"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// SyncSource implements SyncReference with a L2 block sources and L1 hash-by-number source
+type SyncSource struct {
+	L1 eth.BlockLinkByNumber
+	L2 eth.BlockSource
+}
+
+// RefByL1Num fetches the canonical L1 block hash and the parent for the given L1 block height.
```

Comment:
`uint64` is better and more simple, but the standard RPC uses `*big.Int` (also because a nil argument is a special API value for requesting the head).